subsys/net/ip/icmpv4.c

```diff
@@ -26,11 +26,93 @@
 
 static sys_slist_t handlers;
 
+struct net_icmp_hdr *net_icmpv4_set_hdr(struct net_pkt *pkt,
+					struct net_icmp_hdr *hdr)
+{
+	struct net_buf *frag = pkt->frags;
+	struct net_icmp_hdr *icmp_hdr;
+	u16_t pos;
+
+	icmp_hdr = net_pkt_icmp_data(pkt);
+	if (net_check_icmp_header(pkt, icmp_hdr)) {
```

Comment:
These function names are sometimes difficult to invent, please suggest a better name if you have one.

I actually had a comment in the code what it was doing but removed them as they sounded redundant. I can add those back if this section needs clarifications.